pkg/config/setup/config.go

```diff
@@ -365,7 +365,7 @@ func InitConfig(config pkgconfigmodel.Config) {
 	// 	- `statefulsets.apps`
 	// 	- `pods`
 	// 	- `nodes`
-	config.BindEnvAndSetDefault("kubernetes_resources_annotations_as_tags", map[string]map[string]string{})
+	config.BindEnvAndSetDefault("kubernetes_resources_annotations_as_tags", "{}")
```

Comment:
can we add function to validate the format of the entry?

Reply:
@hush-hush pinging to ensure is ok to use `"{}"`

Reply:
This will set the default to a string instead of a `map[string]map[string]string{}`. We're currently working to enforce consistent types across the configuration in order to move toward a schema and better validation of the configuration.

Unless `kubernetes_resources_annotations_as_tags` will be used as a string we want to keep using the real type here.

Reply:
@clamoriniere , @hush-hush 

The issue with using map[string]map[string]string{} is that we don't have an interface method that returns this type from the config. 

For example, we have methods to retrieve map from string to string, and map from string to string slice. (see [here](https://github.com/DataDog/datadog-agent/blob/04d7af75416ceaf46c3bf2be616617285898bd55/pkg/config/model/types.go#L43-L44)). But no method is available to fetch `map[string]map[string]string`. 

While I could've added a method for this, I am not sure it is a good pattern to expose a method for every type (WDYT?). 

In the past we had a similar issue where we wanted to define a config for which the datatype is not very common (it was a list of structs), so we defined it as a json string with default `"[]"` (see [here](https://github.com/DataDog/datadog-agent/blob/7f04426fc9cba8d739bbe59bc9551732dce6e70c/pkg/config/setup/config.go#L770-L773)).

Reply:
To add, if I understand correctly, the config component is built on top of Viper, and Viper doesn't present a method to retrieve the type that I need here. 

To be clear, I am definitely with you regarding enforcing strict types here, but I just can't see how it can be done for the type that I need.